### ISSUE TITLE
Blacklist option ignore generic files without any warning message

### ISSUE THREAD

**Issue by llaville** [box-project/box#945]
## Bug report

| Question    | Answer
| ------------| ---------------
| Box version | 4.3.8@5534406
| PHP version | 8.1.16
| Platform with version | Ubuntu 22.04.2 LTS (GNU/Linux 5.15.90.1-microsoft-standard-WSL2 x86_64)

<details>
 <summary>With the configuration excerpt</summary>
 
 ```json
{
    "blacklist": [
        "*.phar"
    ]
} 
 ```
</details>

![blacklist_generic](https://user-images.githubusercontent.com/364342/226823208-79776f05-8b6b-45b9-ba0f-0c8539552109.png)

PHAR files available into the project source are included into the final PHP Archive compiled, without any warning message that tell that files were ignored !

That lead to a bigger uneccessary PHAR version.

Workaround : we should specify full relative path (to the project) of each individual PHAR files.
 
BTW, it may be  also interresting that the `blacklist` option optionally support contents of  `.gitignore` file.

**Comment by theofidry** [comment]
I don't quite get the problem there, could you rephrase it?

**Comment by llaville** [comment]
With a real example, you should see the problem !

Compare PHAR assets size of https://github.com/llaville/box-manifest/releases/tag/3.0.0 (before optimization) and https://github.com/llaville/box-manifest/releases/tag/3.1.0 (after optimization).

If I let a previous phar built inside the scope of the project source, it will be include into the new PHAR build !
Unless I specify it explicitly :  see https://github.com/llaville/box-manifest/blob/3.1.0/box.json.dist 

But if I try with generic format like this one, it won't works (and previous php archive keep to be included in the new phar generation).

```json
{
    "blacklist": [
        "*.phar"
     ]
}
```

**Comment by llaville** [comment]
Found origin of issue, and also a workaround. I'll try to explain now !

As `blacklist` is not able to accept generic pattern (even if special chars are protected) like this
```json
{
    "blacklist": [
        "*\\.phar"
    ]
}
```
I've read again the official BOX documentation, and play with `finder` setting : https://box-project.github.io/box/configuration/#finder-finder-and-finder-bin

With source code available at : https://github.com/llaville/box-manifest/tree/3.x

Rather than this blacklist : https://github.com/llaville/box-manifest/blob/3.x/box.json.dist#L13-L22
I've tried with a custom `finder` setting like this one 

```json
{
    "finder": [
        {
            "in": ["bin", "src", "vendor"],
            "notName": "*\\.phar",
            "exclude": [
                "fixtures",
                "tests",
                "Test",
                "doc",
                "dist",
                "vendor-bin"
            ]
        }
    ]
}
```
When there are no other `directories` settings it works as expected, the PHAR files inside the my `bin` folder are really ignored.

BUT, and the issue is here, when I've both the `finder` declaration as previous, and a `directories` setting like 
```json
"directories": ["bin", "src", "vendor"]
``` 
it won't work anymore.

The settings from `finder` are ignored and use only the `directories` directive !

The workaround is to remove the `directories` setting and keep the `finder` declaration.

A recommendation or warning message will be greatly helpful in such situation !

**Comment by theofidry** [comment]
>A recommendation or warning message will be greatly helpful in such situation !

I think there is also some similar warnings, so adding one for the finder should be feasible.

Also note that I've always found this finder,include & blacklist settings incredibly annoying and brittle, which is why I added the setting "auto-discovery" which is enabled by default and should be good enough for most cases
